### ISSUE TITLE
gccgo: confusing error message when encountering invalid .. (dot dot) token

### ISSUE THREAD

**Issue by griesemer** [golang/go#28128]
For
```Go
$ cat y.go
package p

func f(x ..int) {}
```
current gccgo reports the errors:
```
$ gccgo -c y.go
y.go:3:10: error: expected package
3 | func f(x ..int) {}
  |          ^
y.go:3:8: error: invalid named/anonymous mix
3 | func f(x ..int) {}
  |        ^
```
I would have expected an error at the `..` as it is not a valid token.

For the reference, cmd/compile reports for the same file:
```
$ go tool compile y.go
y.go:3:11: syntax error: unexpected ., expecting name
```
which seems more sensible.

**Comment by thanm** [comment]
[Not sure what you mean about ".." being an invalid token -- seems like two valid tokens back to back, DOT followed by DOT]

I poked at this a little bit.

What's happening here is that Parse::parameter_list has to guess whether it is parsing a list of just type names, e.g.

   `func(int, int, float64) (*[]int)`

or whether rather a list with parameter names like

   `func(x int, y int, q float64) (*[]int)`

It accepts the identifier "x", then peeks at the next token. Seeing a ".", it decides that this must be a qualified type (e.g. "package.typename") and it proceeds onward from there. It accepts the ".", then tries to look up "x" as a package, which fails -> hence the "expected package" error. This error is issued before we even get to the second dot.

I think it might be possible to get a slightly better error by delaying the package lookup error, however. I'll see what I can come up with.

**Comment by griesemer** [comment]
@thanm You're right; per the spec this should be a DOT followed by a DOT. (The go/scanner consumes the two `.`'s and reports them as a single INVALID token. That behavior though is probably incorrect, even though it may produce a slightly more informative error message, ironically.)
